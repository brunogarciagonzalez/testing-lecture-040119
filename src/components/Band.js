//required modules
import React, { Component } from "react";

class Band extends Component {
  handleOnClick = () => {
    this.props.deleteBand(this.props.id);
  };

  render() {
    return (
      <li>
        {this.props.name}
        <br />
        <button onClick={this.handleOnClick}>Delete</button>
      </li>
    );
  }
}

export default Band;
