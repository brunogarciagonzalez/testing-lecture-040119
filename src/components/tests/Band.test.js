import React from "react";
import { shallow } from "enzyme";
import Band from "../Band";
// 'snapshot testing' : https://jestjs.io/docs/en/snapshot-testing
// 'jest.fn()' : https://jestjs.io/docs/en/jest-object#jestfnimplementation

let mockProps = {
  id: 1,
  name: "Sinatra and the Railers",
  deleteBand: jest.fn()
}

let wrapper = shallow(<Band {...mockProps}/>)

describe("<Band />", ()=> {
  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("deletes band onClick", ()=> {
    wrapper.find("button").simulate("click");

    expect(mockProps.deleteBand).toBeCalledWith(mockProps.id);
  })
})
