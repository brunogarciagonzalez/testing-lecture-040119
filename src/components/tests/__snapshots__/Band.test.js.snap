// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Band /> renders correctly 1`] = `
<li>
  Sinatra and the Railers
  <br />
  <button
    onClick={[Function]}
  >
    Delete
  </button>
</li>
`;
